dockers/docker-dhcp-relay/config.sh

```diff
@@ -1,4 +1,12 @@
 #!/bin/bash
 
-sonic-cfggen -m /etc/sonic/minigraph.xml -y /etc/sonic/dhcp_relay.yml -t /usr/share/sonic/templates/isc-dhcp-relay.j2 > /etc/default/isc-dhcp-relay
+MINIGRAPH_HOSTNAME=`sonic-cfggen -m /etc/sonic/minigraph.xml -v "minigraph_hostname"`
+DEVICE_ROLE=`sonic-cfggen -m /etc/sonic/minigraph.xml -v "minigraph_devices['$MINIGRAPH_HOSTNAME']['type']"`
+
+if [ $DEVICE_ROLE != "ToRRouter" ]; then
+    echo "Device does not require DHCP relay. docker-dhcp-relay exiting..."
```

Comment:
Taoyu is correct. I have tested the behavior, and exiting with a nonzero value will cause the docker container to exit, whereas exiting with a return value of 0 will cause the container to keep running.